src/components/map/components/mobility/Stations.tsx

```diff
@@ -1,67 +1,67 @@
 import React from 'react';
 import {FeatureCollection, GeoJSON} from 'geojson';
-import MapboxGL from '@rnmapbox/maps';
 import {StationBasicFragment} from '@atb/api/types/generated/fragments/stations';
-import {useTheme} from '@atb/theme';
-import {getStaticColor} from '@atb/theme/colors';
-import {getAvailableVehicles} from '@atb/mobility/utils';
 import {FormFactor} from '@atb/api/types/generated/mobility-types_v2';
+import {BikeStations} from './BikeStations';
+import {getAvailableVehicles} from '@atb/mobility/utils';
+import {CarStations} from './CarStations';
 
 type Props = {
   stations: FeatureCollection<GeoJSON.Point, StationBasicFragment>;
 };
 
+export type StationsWithCount = FeatureCollection<
+  GeoJSON.Point,
+  StationBasicFragment & {count: number}
+>;
+
 export const Stations = ({stations}: Props) => {
-  const {themeName} = useTheme();
-  const stationColor = getStaticColor(themeName, 'transport_bike');
+  const getFeaturesOfType = (
+    stations: FeatureCollection<GeoJSON.Point, StationBasicFragment>,
+    formFactor: FormFactor,
+  ) => ({
+    stations: {
+      ...stations,
+      features: stations.features.filter((station) =>
+        station.properties.vehicleTypesAvailable?.some(
+          (type) => type.vehicleType.formFactor === formFactor,
+        ),
+      ),
+    },
+    formFactor,
+  });
 
-  const stationsWithCount = {
+  const stationsWithCount = ({
+    stations,
+    formFactor,
+  }: {
+    stations: FeatureCollection<GeoJSON.Point, StationBasicFragment>;
+    formFactor: FormFactor;
+  }): StationsWithCount => ({
     ...stations,
     features: stations.features.map((feature) => ({
       ...feature,
       properties: {
         ...feature.properties,
         count: getAvailableVehicles(
           feature.properties.vehicleTypesAvailable,
-          FormFactor.Bicycle,
+          formFactor,
         ),
       },
     })),
-  };
+  });
+
+  const bikeStations = stationsWithCount(
+    getFeaturesOfType(stations, FormFactor.Bicycle),
+  );
+  const carStations = stationsWithCount(
+    getFeaturesOfType(stations, FormFactor.Car),
+  );
```

Comment:
I had to read through this code and the functions a couple of times to get a good grasp of what it was doing. Could it be better to combine `stationsWithCount` and `getFeaturesOfType` into a single method? I also have a slight preference for declaring functions outside of the component body if possible.